openfl/plugins/frameworks_adapters/flax_adapter.py

```diff
@@ -0,0 +1,100 @@
+# Copyright (C) 2020-2022 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+"""Custom model DeviceArray - JAX Numpy adapter."""
+import jax
+import jax.numpy as jnp
+import numpy as np
+import optax
+
+from .framework_adapter_interface import FrameworkAdapterPluginInterface
+
+_DELIM = '.'
```

Comment:
If this delimiter is the same for models in Jax/Flax, no need to declare as private global.

Reply:
Used this delimiter to flatten the deeply nested tree of layers, kernel and bias.
Helpful to flatten and unflatten from `get_tensor_dict()` and `set_tensor_dict()` methods respectively (Basically DeviceArray to numpy and vice versa).

Another alternative was to use `flattendict` library. But avoided this import by having a delimiter.

Reply:
Having a delimiter is OK, but if it is well-known, need not declare it as a constant, is what I meant...

Reply:
Sure (y)
I also have an idea to simplify the `get_tensor_dict` and `set_tensor_dict` logic as a whole. This could eliminate the need to have any delimiters. will try and update in the next commit.

Reply:
Refactored this part. 
Improved the logic that handles multiple optax optimizer states.